src/Personal/register.jsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState } from 'react';
+import {
+  View, Text, TextInput, Button,
+} from 'react-native';
+import { useDispatch } from 'react-redux';
+import { useNavigation } from 'react-navigation-hooks';
+import utils from './utils';
+import styles from '../Common/RegisterPage';
+import { requestSignup } from './actionCreator';
+
+export default function Register() {
+  const { navigate } = useNavigation();
+
+  const [username, setUsername] = useState('user1');
+  const [email, setEmail] = useState('1@1.com');
+  const [password, setPassword] = useState('11111111');
+  const [confirmPassword, setConfirmPassword] = useState('11111111');
+
+  const dispatch = useDispatch();
+
+  const handlePasswordChange = (value) => {
+    setPassword(value);
+  };
+  const handleConfirmPasswordChange = (value) => {
+    setConfirmPassword(value);
+  };
+  const handleEmailChange = (value) => {
+    setEmail(value);
+  };
+  const handleUsernameChange = (value) => {
+    setUsername(value);
+  };
+
+  const signUp = (userInfo) => {
+    dispatch(requestSignup(userInfo.email, userInfo.password, userInfo.username));
+    navigate('Index');
```

Comment:
navigate should be invoked after the signup fetch is complete. Use redux state to indicate the the signup status, and monitor the status change and navigate in useEffect()